roles/lib_openshift/src/class/oadm_manage_node.py

```diff
@@ -0,0 +1,208 @@
+    ''' Class to wrap the oc command line tools '''
+
+    # pylint allows 5
+    # pylint: disable=too-many-arguments
+    def __init__(self,
+                 config,
+                 verbose=False):
+        ''' Constructor for OCVolume '''
+        super(ManageNode, self).__init__(None, config.kubeconfig)
+        self.config = config
+
+    def evacuate(self):
+        ''' formulate the params and run oadm manage-node '''
+        return self._evacuate(node=self.config.config_options['node']['value'],
+                              selector=self.config.config_options['selector']['value'],
+                              pod_selector=self.config.config_options['pod_selector']['value'],
+                              dry_run=self.config.config_options['dry_run']['value'],
+                              grace_period=self.config.config_options['grace_period']['value'],
+                              force=self.config.config_options['force']['value'],
+                             )
+    def get_nodes(self, node=None, selector=''):
+        '''perform oc get node'''
+        _node = None
+        _sel = None
+        if node:
+            _node = node
+        if selector:
+            _sel = selector
+
+        results = self._get('node', rname=_node, selector=_sel)
+        if results['returncode'] != 0:
+            return results
+
+        nodes = []
+        items = None
+        if results['results'][0]['kind'] == 'List':
+            items = results['results'][0]['items']
+        else:
+            items = results['results']
+
+        for node in items:
+            _node = {}
+            _node['name'] = node['metadata']['name']
+            _node['schedulable'] = True
+            if 'unschedulable' in node['spec']:
+                _node['schedulable'] = False
+            nodes.append(_node)
+
+        return nodes
+
+    def get_pods_from_node(self, node, pod_selector=None):
+        '''return pods for a node'''
+        results = self._list_pods(node=[node], pod_selector=pod_selector)
+
+        if results['returncode'] != 0:
+            return results
+
+        # When a selector or node is matched it is returned along with the json.
+        # We are going to split the results based on the regexp and then
+        # load the json for each matching node.
+        # Before we return we are going to loop over the results and pull out the node names.
+        # {'node': [pod, pod], 'node': [pod, pod]}
+        # 3.2 includes the following lines in stdout: "Listing matched pods on node:"
+        all_pods = []
+        if "Listing matched" in results['results']:
+            listing_match = re.compile('\n^Listing matched.*$\n', flags=re.MULTILINE)
+            pods = listing_match.split(results['results'])
+            for pod in pods:
+                if pod:
+                    all_pods.extend(json.loads(pod)['items'])
+
+        # 3.3 specific
+        else:
+            # this is gross but I filed a bug...
```

Comment:
I am fixing the current questions.  If drain comes along, I think the plan of action would be to deprecate the old evacuate param in favor of drain. In the code where evacuate is called we could then either check version or we create a lib_openshift that supports the targeted openshift version.